app/controllers/miq_ae_class_controller.rb

```diff
@@ -172,6 +172,9 @@ def get_node_info(node)
       get_method_node_info(id)
     when "aen"
       @record = MiqAeNamespace.find_by_id(from_cid(id[1]))
+      # need to set record as Domain record if it's a domain, editable_domains, enabled_domains,
+      # visible domains methods returns list of Domains, need this for toolbars to hide/disable correct records.
+      @record = MiqAeDomain.find_by_id(from_cid(id[1])) if @record.domain?
```

Comment:
Thanks @h-kataria.  I think the confusion is the multiple classes using the same table without a type column (sti), so activerecord will instantiate the wrong object on the first find above in the case that it is truly a domain.  Another option would be to treat MiqAeNamespace differently from MiqAeDomain by not using "aen" for the latter since clearly their differences are important here.  Why they are treated differently here but not above is really the question?  I don't need to know the answer, just asking questions in case it helps find other ways to treat these classes.